pkg/operator/metrics/metrics.go

```diff
@@ -0,0 +1,161 @@
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/pkg/constants"
+)
+
+const (
+	ControllerName = hivev1.MetricsControllerName
+)
+
+var (
+	// Capture the various conditions set on the HiveConfig
+	metricHiveConfigConditions = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "hive_config_conditions",
+		Help: "HiveConfig with conditions.",
+	}, []string{"condition", "reason"})
+
+	// metricControllerReconcileTime tracks the length of time our reconcile loops take. controller-runtime
+	// technically tracks this for us, but due to bugs currently also includes time in the queue, which leads to
+	// extremely strange results. For now, track our own metric.
+	metricControllerReconcileTime = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "hive_controller_reconcile_seconds",
```

Comment:
Need a better name here, looks like the original one you worked off is pretty bad too, maybe hive_operator_metrics_seconds?

Reply:
It is the time for reconcile loop though - so reconcile in the name seems more appropriate to me - whereas metrics is a bit vague.
I do like the `operator` instead of controller though, since that's how we refer to it generally

Reply:
I agree with @suhanime. I would like to go with `hive_operator_reconcile_seconds`

Reply:
Sounds good to me.